gammapy/datasets/tests/test_map.py

```diff
@@ -1166,6 +1166,58 @@ def test_npred(sky_model, geom, geom_etrue):
         dataset.npred_signal(model_names=["m2"])
 
 
+@requires_data()
+def test_npred_no_cache(sky_model, geom, geom_etrue):
+    import gammapy.datasets.map as dmap
+
+    dmap.USE_NPRED_CACHE = False
+
+    dataset = get_map_dataset(geom, geom_etrue)
+
+    pwl = PowerLawSpectralModel()
+    gauss = GaussianSpatialModel(
+        lon_0="0.0 deg", lat_0="0.0 deg", sigma="0.5 deg", frame="galactic"
+    )
+    model1 = SkyModel(pwl, gauss, name="m1")
+
+    bkg = FoVBackgroundModel(dataset_name=dataset.name)
+    dataset.models = [bkg, sky_model, model1]
+
+    assert_allclose(
+        dataset.npred_signal(model_names=[model1.name]).data.sum(), 150.7487, rtol=1e-3
+    )
+    npred_model1_not_stack = dataset.npred_signal(
+        model_names=[model1.name], stack=False
+    )
+    assert isinstance(npred_model1_not_stack.geom.axes[-1], LabelMapAxis)
+    assert npred_model1_not_stack.geom.axes[-1].name == "models"
+    assert_equal(npred_model1_not_stack.geom.axes[-1].center, [model1.name])
+
+    assert dataset._background_cached is None
+    assert_allclose(dataset.npred_background().data.sum(), 4000.0, rtol=1e-3)
+    assert_allclose(dataset._background_cached.data.sum(), 4000.0, rtol=1e-3)
+
+    assert_allclose(dataset.npred().data.sum(), 9676.047906, rtol=1e-3)
+    assert_allclose(dataset.npred_signal().data.sum(), 5676.04790, rtol=1e-3)
+    assert_allclose(
+        dataset.npred_signal(model_names=[model1.name, sky_model.name]).data.sum(),
+        5676.04790,
+        rtol=1e-3,
+    )
```

Comment:
I am not sure I completely follow the point of the test. Isn't it enough to call `npred()` and then check that the cached maps in the evaluators are correct?

Reply:
yes simplified